File: sample_1_with_java/src/main/java/com/example/sample_1_with_java/ExecuteSQLUtil.java
package com.example.sample_1_with_java;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import java.util.ArrayList;

public class ExecuteSQLUtil {
    public static void insertTodo(Context context, ContentValues values){
        DBHelper helper = new DBHelper(context);
        SQLiteDatabase db = helper.getWritableDatabase();
        db.insert("tb_todo", null, values);
        db.close();
    }
    public static ArrayList<TodoVO> selectTodos(Context context){
        ArrayList<TodoVO> results = new ArrayList<>();
        DBHelper helper = new DBHelper(context);
        SQLiteDatabase db = helper.getReadableDatabase();
        Cursor cursor = db.rawQuery("select * from tb_todo order by date desc", null);
        while (cursor.moveToNext()){
            TodoVO vo = new TodoVO();
            vo.id = cursor.getInt(0);
            vo.title = cursor.getString(1);
            vo.content = cursor.getString(2);
            vo.date = cursor.getString(3);
            vo.completed = cursor.getInt(4);
            results.add(vo);
        }
        db.close();
        return results;
    }

    public static void updateTodo(Context context, ContentValues values, String where){
        DBHelper helper = new DBHelper(context);
        SQLiteDatabase db = helper.getWritableDatabase();
        db.update("tb_todo", values, where, null);
        db.close();
    }

}

File: app/src/main/java/com/example/student/androidlab/ExecuteSQLUtil.kt
package com.example.student.androidlab

import android.content.ContentValues
import android.content.Context

// ExecuteSQLUtil
// contentValues: insert data Map같은 객체
fun insertTodo(context: Context, values: ContentValues) {
    val helper = DBHelper(context)
    val db = helper.writableDatabase // java에선 getWritableDatabase()이지만, 코틀린에선 프로퍼티로 이용 가능하다.

    db.insert("tb_todo", null, values)
    db.close()
}

fun selectTodos(context: Context): MutableList<TodoVO> {
    val result = mutableListOf<TodoVO>()

    val helper = DBHelper(context)
    val db = helper.readableDatabase
    val cursor = db.rawQuery("""select * from tb_todo
        |order by date desc
    """.trimMargin(), null)

    while(cursor.moveToNext()) {
        cursor.run {
            val vo = TodoVO(
                    getInt(0),
                    getString(1),
                    getString(2),
                    getString(3),
                    getInt(4)
                    )
            result.add(vo)
        }
    }
    db.close()
    return result
}